Sources/Error Handling/ErrorCode.swift

```diff
@@ -180,7 +183,12 @@ extension ErrorCode: DescribableError {
             return "This feature is not available when utilizing the customEntitlementsComputation dangerousSetting."
         case .signatureVerificationFailed:
             return "Request failed signature verification. See https://rev.cat/trusted-entitlements for more info."
-
+        case .invalidWebPurchaseToken:
+            return "The link you provided does not contain a valid purchase token."
+        case .alreadyRedeemedWebPurchaseToken:
+            return "The link you provided has already been redeemed."
+        case .expiredWebPurchaseToken:
+            return "The link you provided has expired. A new one will be sent to the customer email."
```

Comment:
This might be more of a backend question but will the email always be sent automatically after trying to redeem an expired token? what if the user has already redeemed a different token for the same purchase?

Reply:
> will the email always be sent automatically after trying to redeem an expired token?

Most times, yes, except when an email has just been sent since it will be throttled, so we don't send continuous emails. cc @antoniobg to confirm.

> what if the user has already redeemed a different token for the same purchase?

The token is associated with a customer actually, also, if the token has already been redeemed we could be facing 2 situations:
- If it was redeemed to an anonymous user id, it will allow further redemptions, and a new one can be generated if expired.
- If it was redeemed to a non-anonymous user id, it will block further redemptions and return a `alreadyRedeemedWebPurchaseToken` so it won't send a new email.

Reply:
> Most times, yes, except when an email has just been sent since it will be throttled, so we don't send continuous emails. cc @antoniobg to confirm.

Yes, basically we only send the email when the token you tried to redeem is expired and matches the latest token we had in file. If you tried to redeem an older token, the email won't be sent. 
I think this can be improved and instead use a rate limit to ensure no more than X emails are sent in X minutes (cc @vicfergar)

Reply:
Thought on the copy: 
> The link you provided has expired. A new one will be sent to the customer email.

99/100 developers are going to just show exactly what we have in the error code message to the end user. It's not ideal, but that's typically what happens. So let's try to write this in a way that it would make sense to an end user and developer alike. 

> The link you provided has expired. A new one will be sent to the email used to make the purchase.